### PR TITLE
refactor: replace hamburger-menu component with theme-ui-menu component

### PR DESCRIPTION
**Pull request by viracoding**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

- In order to have less package dependencies, the hamburger-menu component from a third-party package is replaced with a Button Component (where the icon toggles based on the menu visibility)
- This is actually not really a new feature (but not a bug-fix too). It is just re-implementation of existing functionality.

## Git Issues

- Closes https://github.com/ONEARMY/community-platform/issues/3206

## Screenshots/Videos

Before:
menu-before-open
menu-before-close
After:
menu-after-open
menu-after-close

### CODE REVIEW

File: src/pages/common/Header/Header.tsx [hunk 1]
```diff
@@ -168,16 +167,9 @@ const Header = observer(({ theme }: { theme: ThemeWithName }) => {
...
-              isOpen={isVisible}
-              menuClicked={() => setIsVisible(!isVisible)}
-              width={18}
-              height={15}
-              strokeWidth={1}
-              rotate={0}
-              color="black"
-              borderRadius={0}
-              animationDuration={0.3}
+            <MenuButton
```
Comment (iSCJT) [inline src/pages/common/Header/Header.tsx:170]: I'm not fussed about the animation but could we have an X icon if the menu is open please?

**Review by iSCJT — COMMENTED** [review]
Thanks for this @viracoding.  I'm not fussed about the animation but could we have an X icon if the menu is open please?

**Review by iSCJT — APPROVED** [review]
Thanks for updating @viracoding 👍 

Let's get this merged!

**Comment by viracoding** [comment]
The menu button has just too much spacing on the right side, when comparing with the left side. But it is very difficult to change the styling of the button for a new contributor like me. In addition, in order to add a new button, I had to add code in three different files... 
React-icons are being used as icons, but different icon libraries. This leads to inconsistencies in the long-term. (for example: sSome library uses thinner lines).

**Comment by iSCJT** [comment]
@all-contributors add @viracoding for code

**Pull request merged**